### DISCUSSION TITLE
Setting number of iterations, time step, and convergence criteria

### DISCUSSION THREAD

**Discussion by jundonglin** [PMEAL/OpenPNM#2879] [Q&A]
Hello, how to set the number of iterations, iteration step length and convergence criteria, I can not find the relevant methods to set these on the official website

**Comment by jundonglin** [comment]
I found the example in the script file, I think you should give a guide on the official website, the location of the instance is a little hidden for beginners.

**Comment by jundonglin** [comment]
The code version in the example is too old, and I still can't find the corresponding method for the solver setup

**Comment by ma-sadeghi** [comment]
I assume you're referring to the number of iterations in the nonlinear solver, if that's the case, here's a code snippet from OpenPNM that explains it:

https://github.com/PMEAL/OpenPNM/blob/5e2404c1f39d75064a763bea75c3cd87a6468f7b/openpnm/algorithms/_reactive_transport.py#L21-L46

Basically, once you create your algorithm (`ReactiveTransport` for instance), you can modify these parameters by accessing the algorithm's settings attribute:

```python
alg.settings.newton_maxiter = 100
```

**Reply by ma-sadeghi** [reply]
Upon re-reading the title, it seems that you might be referring to the transient solver? If so, find the corresponding `Settings` class, and you should be able to find relevant parameters there.

**Reply by jundonglin** [reply]
I found it, thank you!
